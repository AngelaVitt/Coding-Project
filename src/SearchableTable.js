import React from 'react';
import ReactTable from 'react-table'
import 'react-table/react-table.css'


const data = [{
      ipAddress: 'Tanner Linsley',
      coordinates: 26,
      friend: {
        name: 'Jason Maurer',
        age: 23,
      }
    }]


    const columns = [{
      Header: 'ipAddress',
      accessor: 'ipAddress' // String-based value accessors!
    }, {
      Header: 'Coordinates',
      accessor: 'coordinates',
      Cell: props => <span className='number'>{props.value}</span> // Custom cell components!
    }, {
      id: 'friendName', // Required because our accessor is not a string
      Header: 'Friend Name',
      accessor: d => d.friend.name // Custom value accessors!
    }, {
      Header: props => <span>Friend Age</span>, // Custom header components!
      accessor: 'friend.age'
    }]
class SearchableTable extends React.Component {
 render() {
    return (
       <ReactTable 
          data={data}
          columns={columns}
          filterable
          defaultPageSize = {10}
           pageSizeOptions = {[5, 10, 20, 25, 50, 100, 500, 1000]}
          defaultFilterMethod={(filter, row, column) => {
    const id = filter.pivotId || filter.id
    return row[id] !== undefined ? String(row[id]).toLowerCase().startsWith(filter.value.toLowerCase()) : true
  }}
          >
        
      </ReactTable>
        
    );
  }
}

export default SearchableTable;
