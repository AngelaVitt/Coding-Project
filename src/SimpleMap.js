import React from 'react';
import GoogleMapReact from 'google-map-react';

import Background from './markerImage.png';

const K_WIDTH = 30;
const K_HEIGHT = 30; 

const markerStyle = {

  position: 'absolute',
  width: K_WIDTH,
  height: K_HEIGHT,
  left: -K_WIDTH / 2,
  top: -K_HEIGHT / 2,

  border: '5px solid #71B300',
  borderRadius: K_HEIGHT,
  backgroundColor: 'white',
  textAlign: 'center',
  color: '#3f51b5',
  fontSize: 16,
  fontWeight: 'bold',
  padding: 4
};

const AnyReactComponent = ({ text }) => (
  
  <img src={Background} style={markerStyle} alt="" />
    
);

export class SimpleMap extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      markers: this.props.markers
    }  
  }

  componentWillReceiveProps(nextProps) {
    this.setState({markers: nextProps.markers});
  }

  static defaultProps = {
    center: {lat: 36.170332, lng: -115.143132},
    zoom: 1
  };

  render() {
    console.log("making map " + JSON.stringify(this.state.markers));
    return (
       <GoogleMapReact
        defaultCenter={this.props.center}
        defaultZoom={this.props.zoom}
      >
        {this.state.markers.map((marker, i) =>{
              return(
                <AnyReactComponent
                  key = {i}
                  lat={marker.lat}
                  lng={marker.lng}
                />

              )
            })}      
      
      </GoogleMapReact>
    );
  }
}
export default SimpleMap;