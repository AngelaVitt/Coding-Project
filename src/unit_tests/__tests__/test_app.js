import GoogleMapReact from 'google-map-react';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import renderer from 'react-test-renderer';

import SearchableTable from './test_searchableTable'
import SimpleMap from './test_simpleMap'

export default class App extends React.Component {
  constructor(props){
      super(props);
      this.state = {
        markers: [{lat: 36, lng: -115}],
        ip_address:[]
      };

      
      this.processData = this.processData.bind(this);
      this.processIPMapping = this.processIPMapping.bind(this);
  }

  render() {
    return(
      <div>
        <div style={{width: '100%', height: '600px'}}>
            <SimpleMap markers={this.state.markers} />
          </div>
          <SearchableTable data={this.state.ip_address} callbackFromParent={this.myCallback} />
        </div>
      );
  }
}


test("renders correctly", () => {
  const renderer = new ShallowRenderer();

  const result = renderer.render(<SimpleMap/>);

  expect(result).toMatchSnapshot();
});


  
