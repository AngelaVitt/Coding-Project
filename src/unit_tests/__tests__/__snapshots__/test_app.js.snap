// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ReactTable
  AggregatedComponent={[Function]}
  ExpanderComponent={[Function]}
  FilterComponent={[Function]}
  LoadingComponent={[Function]}
  NextComponent={undefined}
  NoDataComponent={[Function]}
  PadRowComponent={[Function]}
  PaginationComponent={[Function]}
  PivotComponent={undefined}
  PivotValueComponent={[Function]}
  PreviousComponent={undefined}
  ResizerComponent={[Function]}
  TableComponent={[Function]}
  TbodyComponent={[Function]}
  TdComponent={[Function]}
  TfootComponent={[Function]}
  ThComponent={[Function]}
  TheadComponent={[Function]}
  TrComponent={[Function]}
  TrGroupComponent={[Function]}
  aggregatedKey="_aggregated"
  className=""
  collapseOnDataChange={true}
  collapseOnPageChange={true}
  collapseOnSortingChange={true}
  column={
    Object {
      "Aggregated": undefined,
      "Cell": undefined,
      "Expander": undefined,
      "Filter": undefined,
      "Footer": undefined,
      "Header": undefined,
      "Pivot": undefined,
      "PivotValue": undefined,
      "aggregate": undefined,
      "className": "",
      "filterAll": false,
      "filterMethod": undefined,
      "filterable": undefined,
      "footerClassName": "",
      "footerStyle": Object {},
      "getFooterProps": [Function],
      "getHeaderProps": [Function],
      "getProps": [Function],
      "headerClassName": "",
      "headerStyle": Object {},
      "minWidth": 100,
      "resizable": undefined,
      "show": true,
      "sortMethod": undefined,
      "sortable": undefined,
      "style": Object {},
    }
  }
  columns={
    Array [
      Object {
        "Header": "IP Address",
        "accessor": "ipAddress",
      },
    ]
  }
  data={Array []}
  defaultExpanded={Object {}}
  defaultFilterMethod={[Function]}
  defaultFiltered={Array []}
  defaultPageSize={10}
  defaultResized={Array []}
  defaultSortDesc={false}
  defaultSortMethod={[Function]}
  defaultSorted={Array []}
  expanderDefaults={
    Object {
      "filterable": false,
      "resizable": false,
      "sortable": false,
      "width": 35,
    }
  }
  filterable={true}
  freezeWhenExpanded={false}
  getLoadingProps={[Function]}
  getNoDataProps={[Function]}
  getPaginationProps={[Function]}
  getProps={[Function]}
  getResizerProps={[Function]}
  getTableProps={[Function]}
  getTbodyProps={[Function]}
  getTdProps={[Function]}
  getTfootProps={[Function]}
  getTfootTdProps={[Function]}
  getTfootTrProps={[Function]}
  getTheadFilterProps={[Function]}
  getTheadFilterThProps={[Function]}
  getTheadFilterTrProps={[Function]}
  getTheadGroupProps={[Function]}
  getTheadGroupThProps={[Function]}
  getTheadGroupTrProps={[Function]}
  getTheadProps={[Function]}
  getTheadThProps={[Function]}
  getTheadTrProps={[Function]}
  getTrGroupProps={[Function]}
  getTrProps={[Function]}
  groupedByPivotKey="_groupedByPivot"
  indexKey="_index"
  loading={false}
  loadingText="Loading..."
  multiSort={true}
  nestingLevelKey="_nestingLevel"
  nextText="Next"
  noDataText="No rows found"
  ofText="of"
  onExpandedChange={undefined}
  onFetchData={[Function]}
  onFilteredChange={[Function]}
  onPageChange={[Function]}
  onPageSizeChange={[Function]}
  onResizedChange={undefined}
  onSortedChange={undefined}
  originalKey="_original"
  pageSizeOptions={
    Array [
      5,
      10,
      20,
      25,
      50,
      100,
      500,
      1000,
    ]
  }
  pageText="Page"
  pivotBy={undefined}
  pivotDefaults={Object {}}
  pivotIDKey="_pivotID"
  pivotValKey="_pivotVal"
  previousText="Previous"
  resizable={true}
  rowsText="rows"
  showPageJump={true}
  showPageSizeOptions={true}
  showPagination={true}
  showPaginationBottom={true}
  showPaginationTop={false}
  sortable={false}
  style={Object {}}
  subRowsKey="_subRows"
/>
`;

exports[`renders correctly 2`] = `
<GoogleMap
  debounced={true}
  defaultCenter={undefined}
  defaultZoom={undefined}
  distanceToMouse={[Function]}
  googleMapLoader={[Function]}
  heatmap={Object {}}
  heatmapLibrary={false}
  hoverDistance={30}
  layerTypes={Array []}
  options={[Function]}
  style={
    Object {
      "height": "100%",
      "margin": 0,
      "padding": 0,
      "position": "relative",
      "width": "100%",
    }
  }
  yesIWantToUseGoogleMapApiInternals={false}
/>
`;

exports[`renders correctly 3`] = `
<GoogleMap
  debounced={true}
  defaultCenter={undefined}
  defaultZoom={undefined}
  distanceToMouse={[Function]}
  googleMapLoader={[Function]}
  heatmap={Object {}}
  heatmapLibrary={false}
  hoverDistance={30}
  layerTypes={Array []}
  options={[Function]}
  style={
    Object {
      "height": "100%",
      "margin": 0,
      "padding": 0,
      "position": "relative",
      "width": "100%",
    }
  }
  yesIWantToUseGoogleMapApiInternals={false}
/>
`;
